plugin/src/main/java/org/wildfly/plugin/provision/PackageServerMojo.java

```diff
@@ -318,15 +318,26 @@ private void warnExtraConfig(Path extraContentDir) {
         }
     }
 
-    private Path getDeploymentContent() {
+    private Path getDeploymentContent() throws MojoExecutionException {
         final PackageType packageType = PackageType.resolve(project);
         final String filename;
         if (this.filename == null) {
             filename = String.format("%s.%s", project.getBuild().getFinalName(), packageType.getFileExtension());
         } else {
             filename = this.filename;
         }
-        return targetDir.toPath().resolve(filename);
+        Path deployment = targetDir.toPath().resolve(filename);
+        if (Files.notExists(deployment)) {
+            if (this.filename != null ) {
+                throw new MojoExecutionException("No deployment found wih name " + this.filename);
+            }
+            if (this.runtimeName != null ) {
+                throw new MojoExecutionException("No deployment found although a runtime-name has been set. "
+                        + "Check the final name of the deployment. A custom file name can be set with <filename> parameter.");
+            }
+            getLog().warn("No deployment file found, the server will not contain a deployment");
```

Comment:
Can plugin log, what was he looking for? That mean to log `filename`.

Reply:
@mchoma in this case the filename is null. It can be that no deployment has been set (imagine a pom packaging). This is just a log to advertise that no deployment has been found and deployed.

Reply:
But I see your point, we could still log what we tried to retrieve.